File: 189-rotate-array/189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
      vector<int>z(nums.size(),0);
        int n=nums.size();
        for(int i=0;i<nums.size();i++)
        {
            z[(i+k)%n]=nums[i];

        }
        nums=z;
    }
};

File: 189-rotate-array/189-rotate-array.java
class Solution {
    public void swap(int []nums,int start , int end )
    {
        int temp = nums[start];
        nums[start]=nums[end];
        nums[end]=temp;
    }
    public void reverse(int []nums,int start , int end )
    {
        while(start<=end)
        {
            swap(nums,start,end);
            start++;
            end--;
        }
    }
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k=k%n;
        if(k<0)
            k=k+n;
        reverse(nums,0,n-k-1);
        reverse(nums,n-k,n-1);
        reverse(nums,0,n-1);
    }
}